### DISCUSSION TITLE
[New preprocessor] we added depth leres++

### DISCUSSION THREAD

**Discussion by lllyasviel** [Mikubill/sd-webui-controlnet#1141] [General]
hi everyone, a bit late but we added depth leres++ preprocessor
![image](https://user-images.githubusercontent.com/19834515/235863094-567638b4-09fb-4e93-9603-818755f535e7.png)
Test input:
![image](https://user-images.githubusercontent.com/19834515/235863152-e06d12fb-23c8-4931-8627-45982079a7dc.png)
Midas:
![image](https://user-images.githubusercontent.com/19834515/235863200-f4203621-b54e-4e60-896c-880c6aa7e072.png)
Leres:
![image](https://user-images.githubusercontent.com/19834515/235863245-b0f5ea97-dec2-4205-85ef-f387349e6eef.png)
Zoe:
![image](https://user-images.githubusercontent.com/19834515/235863280-3868f8b8-2613-43ce-9394-c40087aaab98.png)
Leres++:
![image](https://user-images.githubusercontent.com/19834515/235863327-8cca5b17-a6af-40a8-8bd8-206181dac433.png)
You need at least 1.1.133 to use it.

**Comment by danieltudosiu** [comment]
Is there a paper on this leres(++) depth estimator that I can reference/read?

**Comment by TechInterMezzo** [comment]
Same question: Is there any document describing the architecture of leres++? I am not so good at deducing it from the source code.

**Comment by affromero** [comment]
Hey guys, maybe for other folks that come to this discussion trying to understand what "leres++" means. I did not find much information by googling it, so I drop some info here:
https://github.com/Mikubill/sd-webui-controlnet/blob/7fb40bf803d05cde9ca9db8b1c6b8ee3c5efb352/scripts/global_state.py#L56
- `Leres++` is Leres with boosting. 
- What is boosting? I do not use `sd-webui-controlnet`, so I can't tell much about where to find it here, but the first time I came across with the boosting was in the [stable-diffusion-webui-depthmap-script](https://github.com/thygate/stable-diffusion-webui-depthmap-script) repo, and the boosting is referring to this [BoostingMonocularDepth](https://github.com/compphoto/BoostingMonocularDepth) repo, which quoting from there: "We achieve our results by getting several estimations at different resolutions. We then merge these into a structurally consistent high-resolution depth map followed by a local boosting to enhance the results and generate our final result".

Disclaimer: I do not know whether `sd-webui-controlnet` uses the exact same implementation, but at least this can give an idea of what it does and where it comes from.
